indexer/src/types/log.ts

```diff
@@ -47,9 +47,14 @@ export function toEthLog({
   blockHash: PrefixedHexString;
   isPendingBlock: boolean;
 }): JsonRpcLog | null {
-  const { keys, data } = event;
+  const { keys, data, fromAddress } = event;
   const { transactionIndex, hash } = transaction;
 
+  // Log events originated from kakarot address only
+  if (fromAddress !== KAKAROT_ADDRESS) {
```

Comment:
maybe here fromAddress does not start by 0x?

Reply:
Should start with `0x` because event has type `Event` which is

```ts
export declare type Event = {
    /** Event index. */
    index: number;
    /** Contract address. */
    fromAddress: FieldElement;
    /** Event selector. */
    keys: FieldElement[];
    /** Event data. */
    data: FieldElement[];
};
```

and `FieldElement` is :

```ts
type FieldElement = `0x${string}`
```

but I think that in order to have a fair comparison, you should convert both stuff to the same type like it is done in one of our similar function:

```ts
export function isKakarotTransaction(transaction: Transaction): boolean {
  return (
    BigInt(transaction.invokeV1?.calldata?.[1] ?? 0) === BigInt(KAKAROT_ADDRESS)
  );
}
```

Reply:
Thank you for your feedback, I will update the PR with required changes

Reply:
Why are we converting both to `BigInt` here and not `string` given addresses resemble more of a string than integers?

`BigInt(transaction.invokeV1?.calldata?.[1] ?? 0) === BigInt(KAKAROT_ADDRESS)`

Reply:
Because we want to be sure that we are comparing exactly the same things. But as @Eikix said, a string may or may not start with `0x` or other specificities while we know that if we compare two big integers, we are sure of our comparison and its validity.